isso/db/__init__.py

```diff
@@ -114,10 +126,72 @@ def first(rv):
                         ids.extend(rv)
                         flattened[id].update(set(rv))
 
-                for id in flattened.keys():
-                    for n in flattened[id]:
-                        con.execute(
-                            "UPDATE comments SET parent=? WHERE id=?", (id, n))
+                con.execute("BEGIN TRANSACTION")
+                try:
+                    for id in flattened.keys():
+                        for n in flattened[id]:
+                            con.execute(
+                                "UPDATE comments SET parent=? WHERE id=?", (id, n))
+
+                    con.execute('PRAGMA user_version = 3')
+                    con.execute("COMMIT")
+                    logger.info("Database migration to version 3 finished, %i rows changed", con.total_changes)
+                except sqlite3.Error as e:
+                    logger.error("Database migration to version 3 failed: %s", e)
+                    con.execute("ROLLBACK")
+
+        # add "notification" field to "comments" (moved from Comments class to migration)
+        if self.version == 3:
+            with sqlite3.connect(self.path) as con:
+                self.migrate_to_version_4(con)
+
+        # "text" field in "comments" became NOT NULL
+        if self.version == 4:
+            with sqlite3.connect(self.path) as con:
+                con.execute("BEGIN TRANSACTION")
+                con.execute("UPDATE comments SET text = '' WHERE text IS NULL")
+                logger.info("Migrating DB version 4 to 5 by setting empty comments.text to '', %i rows changed",
```

Comment:
Can you please keep the log messages consistent?

Currently a full migration looks like this:
```
migrate database from version 0 to 5
Database migration to version 1 finished, 0 rows changed
Database migration to version 2 finished, 0 rows changed
Database migration to version 3 finished, 0 rows changed
Database migration to version 4 skipped, column 'notification' already exists
Migrating DB version 4 to 5 by setting empty comments.text to '', 0 rows changed
```

Better to emit this log at the end of the migration (after the `COMMIT`):
`Database migration to version 5 finished, %i rows changed by setting empty comments.text to ''`

Also, suggestion for line 71:
```diff
-        logger.info("migrate database from version %i to %i", self.version, to)
+        logger.info("Migrating database from version %i to %i", self.version, to)
```